### PR TITLE
Fix parent close policy processor for deleted namespace

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Fix parent close policy processor for deleted namespace.

**Why?**
Due to bug, `TerminateWorkflow` was still called even if namespace is deleted and not found in namespace registry.

**How did you test it?**
New unit test.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: service/history/transferQueueActiveTaskExecutor.go [hunk 1]
```diff
@@ -1396,15 +1396,17 @@ func (t *transferQueueActiveTaskExecutor) applyParentClosePolicy(
 		return nil
 
 	case enumspb.PARENT_CLOSE_POLICY_TERMINATE:
-		childNamespaceId, err := t.registry.GetNamespaceID(namespace.Name(childInfo.GetNamespace()))
+		childNamespaceID, err := t.registry.GetNamespaceID(namespace.Name(childInfo.GetNamespace()))
 		switch err.(type) {
-		case nil, *serviceerror.NotFound:
+		case nil:
+		case *serviceerror.NotFound:
```
Comment (yycptt) [inline service/history/transferQueueActiveTaskExecutor.go:1402]: Fix the request cancel case below as well?

Comment (alexshtin) [reply service/history/transferQueueActiveTaskExecutor.go:1402]: This is :facepalm:^2.

Comment (alexshtin) [reply service/history/transferQueueActiveTaskExecutor.go:1402]: Updated unit test also.

**Review by yiminc — APPROVED** [review]

**Pull request merged**